xbmc/playlists/PlayListXSPF.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ *  Copyright (C) 2018 Tyler Szabo
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "PlayListXSPF.h"
+
+#include "utils/log.h"
+#include "utils/URIUtils.h"
+#include "utils/XBMCTinyXML.h"
+#include "URL.h"
+
+using namespace PLAYLIST;
+
+/*
+
+Specification: http://www.xspf.org/specs/
+
+Example:
+
+<?xml version="1.0" encoding="UTF-8"?>
+<playlist xmlns="http://xspf.org/ns/0/" version="1">
+	<title>KEXP</title>
+	<trackList>
+		<track>
+			<location>http://live-aacplus-64.kexp.org/kexp64.aac</location>
+			<title>KEXP.ORG 90.3FM - where the music matters</title>
+      <info>https://kexp.org/</info>
+      <image>https://www.kexp.org/static/assets/img/logo-white.svg</image>
+		</track>
+    <track><location>Relative/Path/To.mp3</location></track>
+    <track><location>file:///music/song_1.ogg</location></track>
+    <track><location>file:///music/song_2.flac</location></track>
+    <track><location>file:///music/song_3.mp3</location></track>
+	</trackList>
+</playlist>
+*/
+
+#define FILE_PROTOCOL_WITH_ROOT "file:///"
+#define LOCATION_TAGNAME "location"
+#define PLAYLIST_TAGNAME "playlist"
+#define TITLE_TAGNAME "title"
+#define TRACK_TAGNAME "track"
+#define TRACKLIST_TAGNAME "trackList"
+
+#define missingElementMessage(E) "no " E " element"
+
+
+CPlayListXSPF::CPlayListXSPF(void) = default;
+
+CPlayListXSPF::~CPlayListXSPF(void) = default;
+
+
+bool CPlayListXSPF::Load(const std::string& strFileName)
+{
+  CXBMCTinyXML xmlDoc;
+  const char * errorStr = "";
+  int errorRow = 0;
+  int errorCol = 0;
+
+  const char * label = "";
+  const char * location = "";
+  int offset = 0;
+  std::string localpath;
+  CURL uri;
```

Comment:
This isn't C89 code, so declare the variables where they are actually used.
Also use `std::string` instead of `char*`

Reply:
I use sometimes goto, too. They can help improoving readability

Reply:
Good point on goto being a bit mixed. I'll just refactor to remove them - even if they look clean in this case by having them may encourage attempting to copy the pattern without good reason (and the compiler errors from the uninitialized variables might be painful for someone that's not used to the pattern). It might be handy to add to the style guide to "avoid goto" then I'd just avoid it even if I see the other places.